### ISSUE TITLE
Rerun PR visualizations with iterations info

### ISSUE THREAD

**Issue by crozhon** [gunrock/io#67]
> Finished a lot sooner than I expected, so I added the results to [io/gunrock-output/pr-iteration-rerun/](https://github.com/gunrock/io/tree/master/gunrock-output/pr-iteration-rerun). They have the tag `pr-iteration-rerun` and include the new field `elapsed-iterations`. 
> `elapsed-iterations` is a `num-runs` sized array. If it exists, it specifies the actual number of iterations performed for every run of the problem. On the other hand, the field `max-iter` is an upper bound on the number of iterations performed and doesn't specify how many were executed. 
>Do we have any other programs, besides Pagerank, that need the number of iterations that were executed for a single run?
>_Originally posted by @crozhon in https://github.com/gunrock/gunrock/issues/813#issuecomment-732668635_

Could you try redoing PR with this new `elapsed-iterations` field?

**Comment by jowens** [comment]
@crozhon would like another pair of eyes on this

compare https://gunrock.github.io/docs/#/analysis/results_pr?id=pr-throughput-mteps-for-push-vs-pull-separating-out-options to what's attached

It looks like you only reran K40 with pull and the numbers are considerably worse, which is expected since we are reporting per-run numbers and it looks like we now have 7 iterations not 50.

If that's the case, can you do a full run (V100 and K40 both, push and pull both) and delete all the inaccurate runs in v1-0-0/pr and replace them with the good runs? You can delete pr-iteration-rerun as well. Everything is in source control so we don't lose any info, but the bad files shouldn't be in a current clone of the outputs repo.
[gunrock_primitives_pr_push_pull_mteps.pdf](https://github.com/gunrock/io/files/5598879/gunrock_primitives_pr_push_pull_mteps.pdf)

**Comment by crozhon** [comment]
That's right. The iterations are often considerably smaller than 50, so it makes sense. I only reran with K40 and pull originally, but I started up a full run two nights ago, and it's finishing up today. 

Sounds good. I'll push them and delete the old ones as soon as it's ready.

**Comment by crozhon** [comment]
Do you know why K40C doesn't have any runs with pull=false in the directory? See https://github.com/gunrock/io/tree/master/gunrock-output/v1-0-0/pr/K40c

Did you originally compare Titan V and V100 instead?

**Comment by jowens** [comment]
> Do you know why K40C doesn't have any runs with pull=false in the directory? See https://github.com/gunrock/io/tree/master/gunrock-output/v1-0-0/pr/K40c I'm getting some really long process-times for K40c and the other graphs.
> 
> Did you originally compare Titan V and V100 instead?

I don't have an answer for this. @neoblizz ?

**Comment by jowens** [comment]
[gunrock_primitives_pr_push_pull_mteps.pdf](https://github.com/gunrock/io/files/5600933/gunrock_primitives_pr_push_pull_mteps.pdf)

**Comment by neoblizz** [comment]
> > Do you know why K40C doesn't have any runs with pull=false in the directory? See https://github.com/gunrock/io/tree/master/gunrock-output/v1-0-0/pr/K40c I'm getting some really long process-times for K40c and the other graphs.
> > Did you originally compare Titan V and V100 instead?
> 
> I don't have an answer for this. @neoblizz ?

There was a bug for PR on SM_35 (K40), which is why I never ran a push or pull-based run on K40. The recent pull only results are only there because I ran those for Carl's paper. So, we really need a complete run of push-pull PR on K40.

(Bug may have been https://github.com/gunrock/gunrock/issues/757 or https://github.com/gunrock/gunrock/pull/682)

**Comment by jowens** [comment]
https://github.com/gunrock/io/commit/774bff5a1c89c273bf90292548a15f09ac06a14c
https://github.com/gunrock/io/commit/ed9a5c282b632559fd9f198f396fb043752ea3b2

There are still no K40 push runs, yes @crozhon ?

**Comment by crozhon** [comment]
That's correct. There were still issues running out of memory with push.
